### PR TITLE
Feature/teams

### PR DESCRIPTION
**Pull request by Xx-Parks-xX**

### CODE REVIEW

**Comment by clarsonneur** [comment]
Is it a work in progress, or you are ready to merge it?

File: bitbucket.yaml [hunk 1]
```diff
@@ -149,3 +149,27 @@ objects: # All objects will be delivered by forjj except workspace/infra under o
...
+      repos:
+        help: List of repos separated by comma subscribing to the webhook.
+      team:
+        help: List to enable the webhook at team level. default is false.
+        default: false
+      enabled:
+        help: set 'true' (default) to activate the webhook, 'false' otherwise or 'ignore' to ignore this setup.
+        default: true
+      ssl-check:
+        help: true (default) to ask gitlab to verify the SSL.
```
Comment (clarsonneur) [inline bitbucket.yaml]: gitlab  => bitbucket

File: generated-yaml-structs.go [hunk 2]
```diff
@@ -14,15 +14,15 @@ type AppInstanceStruct struct {
...
 	Key string `json:"key"` // Bitbucket key to access.
-	OrgHookPolicy string `json:"org-hook-policy"` // Set 'sync' to manage all repository webhooks. set 'manage' to manage only listed.
-	OrganizationWebhooksDisabled string `json:"organization-webhooks-disabled"` // true if the plugin should not manage github organization webhooks.
 	ProDeployment string `json:"pro-deployment"` // true if current deployment is production one
 	ProductionTeam string `json:"production-team"` // Production github organization name. By default, it uses the FORJJ organization name
 	ReposDisabled string `json:"repos-disabled"` // true if the plugin should not manage github repositories except the infra repository.
 	ReposWebhooksDisabled string `json:"repos-webhooks-disabled"` // true if the plugin should not manage github repositories webhooks.
 	Secret string `json:"secret"` // Bitbucket secret to access.
 	Server string `json:"server"` // Github Enterprise Server name. By default, public 'github.com' API is used.
 	Team string `json:"team"` // Gitlab group name.
```
Comment (clarsonneur) [inline generated-yaml-structs.go]: Some other reference to gitlab which needs to be replaced

File: generated-yaml-structs.go [hunk 3]
```diff
@@ -14,15 +14,15 @@ type AppInstanceStruct struct {
 	ForjjInfra string `json:"forjj-infra"` // Name of the Infra repository to use in github if requested.
 	ForjjTeam string `json:"forjj-team"` // Default FORJJ group. Used by default as gitlab group. If you want different one, use --gitlab-group
```
Comment (clarsonneur) [inline generated-yaml-structs.go]: gitlab

File: generated-yaml-structs.go [hunk 4]
```diff
@@ -75,6 +75,25 @@ type UserInstanceStruct struct {
...
+
+// Object Instance structures
+
+type WebhooksInstanceStruct struct {
+	Enabled string `json:"enabled"` // set 'true' (default) to activate the webhook, 'false' otherwise or 'ignore' to ignore this setup.
+	Events string `json:"events"` // events requested separated by comma
+	Name string `json:"name"` // webhook name
+	PayloadFormat string `json:"payload-format"` // The media type used to serialize the payloads. Supported values include json and form. The default is form.
+	Repos string `json:"repos"` // List of repos separated by comma subscribing to the webhook.
+	SslCheck string `json:"ssl-check"` // true (default) to ask gitlab to verify the SSL.
```
Comment (clarsonneur) [inline generated-yaml-structs.go]: gitlab

File: generated-yaml-structs.go [hunk 5]
```diff
@@ -308,4 +329,28 @@ const YamlDesc = "---\n" +
...
+   "      repos:\n" +
+   "        help: List of repos separated by comma subscribing to the webhook.\n" +
+   "      team:\n" +
+   "        help: List to enable the webhook at team level. default is false.\n" +
+   "        default: false\n" +
+   "      enabled:\n" +
+   "        help: set 'true' (default) to activate the webhook, 'false' otherwise or 'ignore' to ignore this setup.\n" +
+   "        default: true\n" +
+   "      ssl-check:\n" +
+   "        help: true (default) to ask gitlab to verify the SSL.\n" +
```
Comment (clarsonneur) [inline generated-yaml-structs.go]: another gitlab

File: bitbucket_plugin.go [hunk 6]
```diff
@@ -4,54 +4,53 @@
...
-type BitbucketDeployStruct struct{
-	goforjj.PluginService 	`yaml:",inline"` //urls
-	Repos 					map[string]RepositoryStruct // projects managed in gitlab
-	NoRepos 				bool `yaml:",omitempty"`
-	ProdTeam 				string
-	Team 					string
-	TeamDisplayName 		string
+type BitbucketDeployStruct struct {
+	goforjj.PluginService `yaml:",inline"`            //urls
+	Repos                 map[string]RepositoryStruct // projects managed in gitlab
```
Comment (clarsonneur) [inline bitbucket_plugin.go]: gitlab

**Review by clarsonneur — CHANGES_REQUESTED** [review]
Several reference to gitlab to update

I guess, the code is still in progress. Think to add/update your README.md to any extra information, if needed.

**Comment by Xx-Parks-xX** [comment]
The code has been cleaned, the branch can be merged.

**Pull request merged**